The Slime Update (Part 1)

paige404 opened the pull request:
## About the PR
<!-- What did you change in this PR? -->
<!-- Что вы изменили в своем пулл реквесте? -->
As the slimes and royal jelly stuff felt like it was scope creeping too much, I have decided to split it into smaller portions. This PR constitutes the general slime changes and some preparatory reorganizing for Royal Jelly.

I gave different damage modifiers to the different slime varieties, and nerfed the electric slimes a little bit. I added a new damage container: CP14BiologicalAnaerobic for creatures that are (mostly) biological but won't suffocate, currently only used by slimes. I also adjusted the damage taken at high and low temperatures by slimes.

~~This also adds a pet slime, Opus, designed to be an alchemist's familiar. The first "station" pet of Crystal Edge, and hopefully not the last. Opus won't be hurt by other slimes, but WILL be attacked by other kinds of monsters. Watch out for the little guy, as he will gib on death just like any other slime. Opus has a unique sprite, but it is just a purple recolor of the base slime for now; I want time to collect feedback for any distinguishing changes people want to see on him. Opus was developed in collaboration with muichiro__toki on discord.~~

Finally, slime crown has been properly given its DO NOT MAP tag.

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
<!-- Зачем нужно это изменение? Прикрепите любые обсуждения или проблемы здесь. Опишите, как это повлияет на текущий баланс игры. -->
It made no sense for ice slimes to be hurt by cold and fire slimes to be hurt by fire. Everyone found electric slimes annoying--they're still annoying, but should be more manageable with much less risk of inescapable stunlock.

## Media
<!--
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
-->
<!--
Пулл реквесты, которые несут за собой игровые изменения (добавления одежды, предметов и так далее) требуют чтобы вы прикрепили скриншоты или видеоролики, демонстрирующие эти изменения.
Небольшие исправления не считаются.
-->

https://github.com/user-attachments/assets/d03c298d-743c-4d5b-ad97-bc3ef6206a99

## Changelog
Most of this is functionally invisible to the end user, so we'll just CL what the user will notice

:cl:
- tweak: Tweaked slime damage resistances and immunities. Elemental slimes are now immune to their own elements, but weak to an opposing element. Electric slimes take equal damage from both heat and cold, and have had their oppressive stunlocking power somewhat reduced. Additionally, all slimes are now immune to airloss damage.

KittyCat432 commented:
Blyump 👍

Nimfar11 commented on Resources/Prototypes/_CP14/ai_factions.yml:
```diff
@@ -17,6 +19,8 @@
   - CP14Monster
   - CP14Slimes
+  - CP14Pets
+  - CP14HostileEveryone
```
Are they hostile to themselves?

paige404 replied:
CP14HostileEveryone doesn't get used by anything currently. I just thought it would make sense, if we use that ever, for them to be hostile against their own faction too.

Granted it is outside the scope of this PR, I'm happy to strike that out of it.

Nimfar11 replied:
We should probably put it away for now. In the future there may be a mob with this faction, and it can be confusing if it attacks everyone, including other mobs with the same faction.

paige404 replied:
It is done

Nimfar11 commented:
You need to create a spawner on the pet so that mappers can add them to the map.

Nimfar11 reviewed (approved):
There don't seem to be any errors.

paige404 commented:
> And why make a purple copy of the sprite when you can color through prototypes?

I anticipate further changes to the sprite in the near future based on player feedback. Also, Spiro didn't like the eye color when I just recolored it in the yaml.

paige404 commented:
If it's a dealbreaker I can shelve the sprite for now and just use the prototype to color it. Spiro just thought the eyes were too dull when I did that, and I suspect players are going to want some additional feature to distinguish Opus from other slimes anyway.

paige404 commented on Resources/Locale/en-US/_CP14/ghostRoles/roles.ftl:
```diff
@@ -1,8 +1,11 @@
+cp14-ghost-role-information-opus-name = Opus
+cp14-ghost-role-information-opus-description = The alchemist's haughty familiar. A purple slime of princely bearing. Help the alchemist, but play nice with other slimes.
+
```
Oh! Okay. I hope the translator doesn't mess it up too much.

paige404 commented on Resources/Prototypes/_CP14/Entities/Mobs/NPC/Slimes/fire.yml:
```diff
@@ -21,19 +21,21 @@
           Quantity: 20
         - ReagentId: CP14EssenceFire
-          Quantity: 20
-        - ReagentId: CP14BasicEffectDamageHeat
```
Oh, you're right. For some reason I thought the fire essence was setting the fires. I'll put it back.

paige404 commented on Resources/Prototypes/_CP14/Entities/Mobs/NPC/pets.yml:
```diff
@@ -0,0 +1,103 @@
+  - type: Grammar
+    attributes:
+      proper: true
+      gender: neuter
```
A few things are duplicates, you're right. But many are actual deviations from the parent. Opus is a proper noun while slime is not (my studies in Russian have so far not shown that this is a meaningful distinction in Russian though). Opus has different damage and destruction thresholds from the parent, and weaker melee attacks. Opus has quieter steps to signify that it isn't a threat.

With extensive trial and error I have whittled out all of the excess on my branch though, and everything that remains when I update this PR is a deliberate change from the parent base slime.

paige404 replied:
Alright, I'm ditching the footstep sounds because I can't actually hear a difference in the volume levels.

paige404 commented on Resources/Prototypes/_CP14/Entities/Mobs/NPC/pets.yml:
```diff
@@ -0,0 +1,103 @@
+      - !type:DoActsBehavior
+        acts: [ "Destruction" ]
+  # By request, we can feed this slime (no starvation damage though; that's annoying on a pet)
+  - type: Hunger
```
On further testing, I don't think the slime is ingesting the food and water even with hunger and thirst. I think I would need to give it a stomach? Regardless, it's more headache than it's worth. We're ditching the hunger for now.

paige404 commented:
Following discussion with both Ed and ayou, I have removed Opus from the PR. He exists on his own branch now, to be PRed when we have a clearer concept on the role of pets in this server.

KittyCat432 commented:
> Following discussion with both Ed and ayou, I have removed Opus from the PR. He exists on his own branch now, to be PRed when we have a clearer concept on the role of pets in this server.

:<

paige404 commented:
Okay, updated with a new video showcasing the slimes surviving indefinitely in the water and dying at different speeds in lava (except the fire slime, who is happy and comfy).

paige404 commented:
> Due to an update to the changelog system, please make sure your changelog description correctly describes changes that affect player gameplay:
> 
> ```
> :cl:
> - add: Added fun!
> - remove: Removed fun!
> - tweak: Changed fun!
> - fix: Fixed fun!
> ```

Done!

TheShuEd commented:
I think :cl: should be more comprehensive, considering that there are many important balance adjustments here.

TheShuEd commented:
Thats good!
